lib/websocket.js

```diff
@@ -511,6 +519,9 @@ function initAsClient (address, protocols, options) {
   }
   if (options.host) requestOptions.headers.Host = options.host;
   if (serverUrl.auth) requestOptions.auth = serverUrl.auth;
+  else if (serverUrl.username && serverUrl.password) {
```

Comment:
Still not enough I think:

```
var u = new url.URL('ws://foo@example.org')
undefined
> u
URL {
  href: 'ws://foo@example.org/',
  origin: 'ws://example.org',
  protocol: 'ws:',
  username: 'foo',
  password: '',
  host: 'example.org',
  hostname: 'example.org',
  port: '',
  pathname: '/',
  search: '',
  searchParams: URLSearchParams {},
  hash: '' }
```

`auth` should be `'foo'` but it doesn't get set as `serverUrl.password` is falsy.